# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1467742075 25200
#      Tue Jul 05 11:07:55 2016 -0700
# Node ID 0a528a6dac958c19cda37301623c9408856727f7
# Parent  a4eb6d3de9f0d1543659dcb1cab9b7d03625ec61
build: use pkg-config in OCTAVE_CHECK_LIB without overriding user options

* acinclude.m4 (OCTAVE_CHECK_LIB): Use pkg-config only when configure is not
called with --with-LIB or --without-LIB options.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -644,39 +644,47 @@ AC_DEFUN([OCTAVE_CHECK_LIB], [
   AC_SUBST(m4_toupper([$1])_LDFLAGS)
 
   AC_ARG_WITH([$1],
     [m4_ifblank([$7],
       [AS_HELP_STRING([--without-$1], [don't use $2 library])],
       [AS_HELP_STRING([--without-$1], [$7])])],
     with_$1=$withval, with_$1=yes)
 
+  ac_octave_$1_pkg_check=no
   m4_toupper([$1])_LIBS=
   warn_$1="$3"
   case $with_$1 in
     no)
       warn_$1="--without-$1 specified.  Functions or features that depend on $2 will be disabled."
       m4_toupper([$1])_LIBS=
     ;;
     yes | "")
+      ac_octave_$1_pkg_check=yes
       m4_toupper([$1])_LIBS="-l$1"
     ;;
     -* | */* | *.a | *.so | *.so.* | *.o)
       m4_toupper([$1])_LIBS="$with_$1"
     ;;
     *)
       m4_toupper([$1])_LIBS="-l$with_$1"
     ;;
   esac
 
-  PKG_CHECK_EXISTS([$1], [
-    m4_toupper([$1])_CPPFLAGS="$($PKG_CONFIG --cflags-only-I $1) $m4_toupper([$1])_CPPFLAGS"
-    m4_toupper([$1])_LDFLAGS="$($PKG_CONFIG --libs-only-L $1) $m4_toupper([$1])_LDFLAGS"
-    m4_toupper([$1])_LIBS="$($PKG_CONFIG --libs-only-l $1) $m4_toupper([$1])_LIBS"
-  ])
+  if test $ac_octave_$1_pkg_check = yes; then
+    PKG_CHECK_EXISTS([$1], [
+      if test -z "$m4_toupper([$1])_CPPFLAGS"; then
+        m4_toupper([$1])_CPPFLAGS="$($PKG_CONFIG --cflags-only-I $1)"
+      fi
+      if test -z "$m4_toupper([$1])_LDFLAGS"; then
+        m4_toupper([$1])_LDFLAGS="$($PKG_CONFIG --libs-only-L $1)"
+      fi
+      m4_toupper([$1])_LIBS="$($PKG_CONFIG --libs-only-l $1)"
+    ])
+  fi
 
   if test -n "$m4_toupper([$1])_LIBS"; then
     ac_octave_save_CPPFLAGS="$CPPFLAGS"
     ac_octave_save_LDFLAGS="$LDFLAGS"
     ac_octave_save_LIBS="$LIBS"
     CPPFLAGS="$m4_toupper([$1])_CPPFLAGS $CPPFLAGS"
     LDFLAGS="$m4_toupper([$1])_LDFLAGS $LDFLAGS"
     LIBS="$m4_toupper([$1])_LIBS $LIBS"
